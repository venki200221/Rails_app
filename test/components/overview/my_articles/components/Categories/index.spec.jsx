import React from 'react';
import { shallow } from 'enzyme';
import '../../../../../testHelper';

import Categories from '@components/overview/my_articles/components/Categories/Categories';

describe('Categories', () => {
  const props = {
    assignments: [],
    course: { home_wiki: { project: 'wikipedia' } },
    current_user: {},
    wikidataLabels: {}

  };

  it('should show articles the user is creating', () => {
    const creatingProps = {
      ...props,
      assignments: [
        { article_status: 'new_article' }
      ]
    };
    const component = shallow(<Categories {...creatingProps} />);
    expect(component).toMatchSnapshot();
  });

  it('should show articles the user is improving', () => {
    const creatingProps = {
      ...props,
      assignments: [
        { article_status: 'improving_article' }
      ]
    };
    const component = shallow(<Categories {...creatingProps} />);
    expect(component).toMatchSnapshot();
  });

  it('should show articles the user is reviewing', () => {
    const creatingProps = {
      ...props,
      assignments: [
        { article_status: 'reviewing_article' }
      ]
    };
    const component = shallow(<Categories {...creatingProps} />);
    expect(component).toMatchSnapshot();
  });
});
