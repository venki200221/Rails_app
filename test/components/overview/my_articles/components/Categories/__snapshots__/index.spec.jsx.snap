// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories should show articles the user is creating 1`] = `ShallowWrapper {}`;

exports[`Categories should show articles the user is improving 1`] = `ShallowWrapper {}`;

exports[`Categories should show articles the user is reviewing 1`] = `ShallowWrapper {}`;
